### PR TITLE
Add more functional properties for comparisons

### PR DESCRIPTION
**Pull request by leocavalcante**
### Next
Implement all comparisons operators
http://php.net/manual/en/language.operators.comparison.php

### CODE REVIEW

**Comment by leocavalcante** [comment]
After writing example on #123 I'm willing to rename `cond` to `match`, what you think?

File: src/Functional/Always.php [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+namespace Functional;
+
+/**
+ * Always returns the given value. Like `Functional\id`, but it is a function
+ *
+ * @param mixed $x the value to always return
+ *
+ * @return mixed the given always return value
+ */
+function always($x)
```
Comment (lstrojny) [inline src/Functional/Always.php]: This one exists already and is called `const_function` right now. We could have an alias but I am not sure we need one.

File: src/Functional/Cond.php [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+use Functional\always;
+
+/**
+ * Performs an operation checking for the given conditions
+ *
+ * @param array $conds the conditions to check against
+ *
+ * @return callable|null the function that calls the callable of the first truthy condition
+ */
+function cond(array $conds)
```
Comment (lstrojny) [inline src/Functional/Cond.php]: Let’s use the full name, `$conditions`. Also the function should be called `condition`

File: src/Functional/IfElse.php [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+/**
+ * Performs an if/else condition over a value using functions as statements
+ *
+ * @param callable $cond the condition function
+ * @param callable $if   function to call if condition is true
+ * @param callable $else function to call if condition is false
+ *
+ * @return mixed the return value of the given $if or $else functions
+ */
+function if_else(callable $cond, callable $if, callable $else)
```
Comment (lstrojny) [inline src/Functional/IfElse.php]: What about changing the argument names to be `callable $if, callable $then, callable $else`

File: src/Functional/Cond.php (hunk 2 above)
Comment (lstrojny) [inline src/Functional/Cond.php]: Thinking about the name here, it seems more like a value match then a condition. What about `match` as a function name?

Comment (lstrojny) [inline src/Functional/Cond.php]: Haven’t seen your comment on using `match` just now. So I guess we have an agreement :)

**Comment by lstrojny** [comment]
Looks good. Can you add documentation for it?

**Comment by leocavalcante** [comment]
Done!

File: docs/functional-php.md [hunk 4]
```diff
@@ -878,6 +881,78 @@ Returns a unified array based on the index value returned by the callback, use `
...
+`$conditions` should be a bi-dimensional array with items following the given signature: `[callable $if, callable $then]`.
+`$if` is the predicate, that when returns a truthy value `$then` is called.
+It stops on the first match and if none of the conditions matches, `null` is returned.
+
+```php
+<?php
+
+use function Functional\match;
+
+$above = function ($limit) {
```
Comment (lstrojny) [inline docs/functional-php.md]: you can use the greater_than_or_equals function here

File: docs/functional-php.md [hunk 5]
```diff
@@ -878,6 +881,78 @@ Returns a unified array based on the index value returned by the callback, use `
...
+``callable if_else(callable $if, callable $then, callable $else)``
+Returns a new function that will call `$then` if the return of `$if` is truthy, otherwise calls `$else`.
+All three functions will be called with the given argument.
+
+```php
+<?php
+
+use function Functional\if_else;
+
+$isBiggerThan1 = function ($value) {
```
Comment (lstrojny) [inline docs/functional-php.md]: You can use the greater_than function here

File: src/Functional/Match.php [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+ * @return callable|null the function that calls the callable of the first truthy condition
+ */
+function match(array $conditions)
+{
+    return function ($value) use ($conditions) {
+        if (empty($conditions)) {
+            return null;
+        }
+
+        list($if, $then) = head($conditions);
```
Comment (lstrojny) [inline src/Functional/Match.php]: Do we want to check if the array actually contains exactly two elements and both are callable?

**Comment by leocavalcante** [comment]
Hi, @lstrojny - I wanted to avoid using another lib function while explaining one, but if it is ok... Also, didn't find any satisfiable solution to use already implemented assertions with match() needs, so I created another one following the same approach, what you think?

**Comment by lstrojny** [comment]
Looks very good now! Thank you

**Pull request merged**